src/components/BMDashboard/Equipment/DailyActivityLog/EDailyActivityLog.jsx

```diff
@@ -0,0 +1,328 @@
+
+    const allNumbers = buildToolNumbers(e.itemType?.name, total);
+    const inUseNumbers = allNumbers.slice(0, usingQty);
+    const availableNumbers = allNumbers.slice(usingQty);
+
+    return {
+      id: e._id,
+      name: e.itemType?.name || 'Unknown',
+      workingQty: total,
+      usingQty,
+      availableQty,
+      inUseNumbers,
+      availableNumbers,
+      selectedNumbers: [],
+    };
+  });
+
+function EDailyActivityLog(props) {
+  const dispatch = useDispatch();
+
+  /* redux slices */
+  const bmProjects = useSelector(s => s.bmProjects || []);
+  const equipments = useSelector(s => s.bmEquipments?.equipmentslist || []);
+  const darkMode = useSelector(state => state.theme.darkMode);
+
+  const { user } = props.auth;
+
+  /* local state */
+  const [selectedProject, setSelectedProject] = useState(null);
+  const [date, setDate] = useState(TODAY);
+  const [logType, setLogType] = useState('check-in'); // 'check-in' | 'check-out'
+  const [rows, setRows] = useState([]);
+
+  useEffect(() => {
+    dispatch(fetchBMProjects());
+  }, [dispatch]);
+
+  /* fetch equipments whenever project changes */
+  useEffect(() => {
+    if (selectedProject?.value) {
+      dispatch(fetchAllEquipments(selectedProject.value));
+    }
+  }, [selectedProject, dispatch]);
+
+  /* build rows whenever equipments slice updates */
+  const derived = useMemo(() => buildRows(equipments), [equipments]);
+  useEffect(() => setRows(derived), [derived]);
+
+  const onToolSelect = (rowIdx, selected) => {
+    setRows(prev => {
+      const row = prev[rowIdx];
+      const valid = logType === 'check-in' ? row.availableNumbers : row.inUseNumbers;
+      const limit = logType === 'check-in' ? row.availableQty : row.usingQty;
+
+      const clean = selected
+        .map(o => o.value)
+        .filter(v => valid.includes(v))
+        .slice(0, limit);
+
+      return prev.map((r, i) => (i === rowIdx ? { ...r, selectedNumbers: clean } : r));
+    });
+  };
+
+  const flipLogType = newType => {
+    setLogType(newType);
+    setRows(prev =>
+      prev.map(r => {
+        const valid = newType === 'check-in' ? r.availableNumbers : r.inUseNumbers;
+        const limit = newType === 'check-in' ? r.availableQty : r.usingQty;
+        return {
+          ...r,
+          selectedNumbers: r.selectedNumbers.filter(n => valid.includes(n)).slice(0, limit),
+        };
+      }),
+    );
+  };
+
+  const handleCancel = () => {
+    setSelectedProject(null);
+    setRows([]);
+    setLogType('check-in');
+    setDate(TODAY);
+  };
+
+  const handleSubmit = () => {
+    const payload = rows.flatMap(r =>
+      r.selectedNumbers.map(() => ({
+        equipmentId: r.id,
+        logEntry: {
+          createdBy: user.userid,
+          responsibleUser: null,
+          type: logType === 'check-in' ? 'Check In' : 'Check Out',
+          date,
+        },
+      })),
+    );
+
+    dispatch(updateMultipleEquipmentLogs(selectedProject.value, payload));
```

Comment:
It seems 'selectedNumbers' are not getting passed in the request payload, leading to inconsistent values when toggling between 'Check-in' and 'Check-out'.